src/Psalm/Internal/Codebase/Functions.php

```diff
@@ -555,15 +555,15 @@ public function isCallMapFunctionPure(
             return true;
         }
 
-        if ($function_id === 'count' && isset($args[0]) && $type_provider) {
+        if (in_array($function_id, ['count', 'sizeof']) && isset($args[0]) && $type_provider) {
             $count_type = $type_provider->getType($args[0]->value);
 
             if ($count_type) {
                 foreach ($count_type->getAtomicTypes() as $atomic_count_type) {
                     if ($atomic_count_type instanceof TNamedObject) {
                         $count_method_id = new MethodIdentifier(
                             $atomic_count_type->value,
-                            'count',
+                            $function_id,
```

Comment:
Here we're checking if `ClassName::count()` (an implementation of `Countable`) has side-effects. Since both `count` and `sizeof` will still call `ClassName::count()`, this should stay constant

```suggestion
                            'count',
```

Reply:
Got it, changed :)